### PR TITLE
ICU-20438 API status promotion for ICU4J 64

### PR DESCRIPTION
**Pull request by yumaoka**
Also includes API tag fixes. build.properties was updated for producing API change report.

##### Checklist

- [x] Issue filed: https://unicode-org.atlassian.net/browse/ICU-20438
- [x] Updated PR title and link in previous line to include Issue number
- [x] Issue accepted
- [ ] Tests included
- [ ] Documentation is changed or added

### CODE REVIEW

File: icu4j/main/classes/core/src/com/ibm/icu/text/DateTimePatternGenerator.java [hunk 1]
```diff
@@ -400,7 +400,7 @@ public void put(UResource.Key key, UResource.Value value, boolean noFallback) {
     // Get the data for dayperiod C.
     static final Map<String, String[]> LOCALE_TO_ALLOWED_HOUR;
     static {
-        HashMap<String, String[]> temp = new HashMap<String, String[]>();
+        HashMap<String, String[]> temp = new HashMap<>();
```
Comment (yumaoka) [inline icu4j/main/classes/core/src/com/ibm/icu/text/DateTimePatternGenerator.java]: Automatically updated by Eclipse when I touched the file. I'd leave them as is.

File: icu4j/main/classes/core/src/com/ibm/icu/util/Currency.java [hunk 2]
```diff
@@ -236,7 +235,6 @@ public static Currency getInstance(ULocale locale) {
         return resultSet;
     }
 
-    private static final String EUR_STR = "EUR";
```
Comment (yumaoka) [inline icu4j/main/classes/core/src/com/ibm/icu/util/Currency.java]: Fixed unused warning.

File: icu4j/main/classes/core/src/com/ibm/icu/text/ConstrainedFieldPosition.java [hunk 3]
```diff
@@ -193,8 +193,10 @@ public void constrainClass(Class<?> classConstraint) {
      *
      * @param field The field to fix when iterating.
      * @param fieldValue The field value to fix when iterating.
-     * @internal ICU 64 Technical Preview
+     * @internal ICU 64 technology preview
+     * @deprecated This API is for technology preview and might be changed or removed in a future release.
```
Comment (yumaoka) [inline icu4j/main/classes/core/src/com/ibm/icu/text/ConstrainedFieldPosition.java]: API tool requires `@deprecated` is used along with `@internal`. The description of `@internal` should contain the phrase 'technology preview', that is recognized by the tool. Although, we may revisit this to change the restriction after 64.

File: icu4j/main/classes/core/src/com/ibm/icu/text/ConstrainedFieldPosition.java [hunk 4]
```diff
@@ -344,6 +346,11 @@ public boolean matchesField(Field field, Object fieldValue) {
         }
     }
 
+    /**
+     * {@inheritDoc}
+     * @draft ICU 64
+     * @provisional This API might change or be removed in a future release.
+     */
```
Comment (yumaoka) [inline icu4j/main/classes/core/src/com/ibm/icu/text/ConstrainedFieldPosition.java]: Even `@Override`, we still need API status tag. In this case, the API status should match the class's status.

File: icu4j/APIChangeReport.html [hunk 5]
```diff
@@ -2,87 +2,112 @@
 <html>
 <head>
 <meta http-equiv="Content-Type" content="text/html; charset=utf-8">
-<!-- © 2018 and later: Unicode, Inc. and others. -->
+<!-- © 2019 and later: Unicode, Inc. and others. -->
 <!-- License & terms of use: http://www.unicode.org/copyright.html#License -->
-<title>ICU4J API Comparison: ICU4J 62.1 with ICU4J 63.1</title>
+<title>ICU4J API Comparison: ICU4J 63.1 with ICU4J 63.1</title>
```
Comment (pedberg-icu) [inline icu4j/APIChangeReport.html]: API Comparison: ICU4J 63.1 with ICU4J 63.1</title>

Comparing 63 with 63??

**Comment by pedberg-icu** [comment]
> LGTM

Actually I had not reviewed APIChangeReport.html, it needs some changes

File: icu4j/APIChangeReport.html (hunk 5 above)
Comment (yumaoka) [reply icu4j/APIChangeReport.html]: Oops.. I'll update.

File: icu4j/main/classes/core/src/com/ibm/icu/text/ConstrainedFieldPosition.java (hunk 4 above)
Comment (markusicu) [reply icu4j/main/classes/core/src/com/ibm/icu/text/ConstrainedFieldPosition.java]: I am always confused about API status of overrides. I thought we had said recently that they should match the status of what they inherit, but when the whole class is new using the same status as the class makes sense... so ok.

**Pull request merged**